ha_mqtt_discoverable/__init__.py

```diff
@@ -553,21 +557,22 @@ class Discoverable(Generic[EntityType]):
     mqtt_client: mqtt.Client
     wrote_configuration: bool = False
     # MQTT topics
-    topic_prefix: str
+    _entity_topic: str
     config_topic: str
     state_topic: str
     availability_topic: str
 
-    class Config:
-        arbitrary_types_allowed = True
-
-    def __init__(self, settings: Settings[EntityType]) -> None:
+    def __init__(
+        self, settings: Settings[EntityType], on_connect: Optional[Callable] = None
+    ) -> None:
         """
-        Setup the base discoverable object class
+        Creates a basic discoverable object.
 
         Args:
-            settings: Settings for the sensor we want to create in Home Assistant. \
-                See the `Settings` class for the available options.
+            settings: Settings for the entity we want to create in Home Assistant.
```

Comment:
I'm going to formalize in `Contributing.md` that we use google-style docstrings like this in the module.

Reply:
Well, If you prefer some other style, please let me know. It's just a style that comes to me naturally when writing my projects nowadays. :sweat_smile: